lib/global_id/global_id.rb

```diff
@@ -16,6 +19,9 @@ def create(model, options = {})
         raise ArgumentError, 'An app is required to create a GlobalID. ' \
           'Pass the :app option or set the default GlobalID.app.'
       end
+    rescue URI::InvalidComponentError
```

Comment:
Would prefer having this in the `URI::GID` class. Change the `model_id` validation here: https://github.com/rails/globalid/blob/master/lib/global_id/uri/gid.rb#L144